pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -347,7 +347,7 @@ func (c *Reconciler) reconcileDeployment(el *v1alpha1.EventListener) error {
 		el.Status.SetDeploymentConditions(existingDeployment.Status.Conditions)
 		// Determine if reconciliation has to occur
 		updated := reconcileObjectMeta(&existingDeployment.ObjectMeta, deployment.ObjectMeta)
-		if existingDeployment.Spec.Replicas == nil || *existingDeployment.Spec.Replicas == 0 {
+		if existingDeployment.Spec.Replicas != deployment.Spec.Replicas {
```

Comment:
```suggestion
		if existingDeployment.Spec.Replicas == nil ||  existingDeployment.Spec.Replicas != deployment.Spec.Replicas {
```
I feel it should be above. It's quite possible for replica in deployment to be nil, then reconciler will panic. What if user manually changed deployment's replica to zero or empty by supplying yaml file?

Reply:
@khrm replica in deployment will never be nil because we always set replicas to `1` https://github.com/tektoncd/triggers/blob/master/pkg/reconciler/v1alpha1/eventlistener/eventlistener.go#L262 so considering that we can assume like `existingDeployment.Spec.Replicas` will never be nil

Reply:
What if user set it nil in Deployment YAML bypassing EL?

Reply:
setting `replicas` to **nil**  by `kubectl edit deployment` is not allowed which gives error like below
```
* : Invalid value: "The edited file failed validation": ValidationError(Deployment.spec.replicas): invalid type for io.k8s.api.apps.v1.DeploymentSpec.replicas: got "string", expected "integer"
```
and if we set `replicas` to 0 by doing `kubectl edit deployment`  it will rewrite to `1`

Reply:
You can set nil by not giving or removing `replicas` during `kubectl edit deployment`.
But I check that defaults would initialize the pointer and put it to 0 if its nil. https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/pkg/apis/apps/v1beta1/defaults.go#L85